### ISSUE TITLE
mechanism of providing a path to ic-ajax is no compatible with NPM

### ISSUE THREAD

**Issue by stefanpenner** [rwjblue/ember-cli-ic-ajax#6]
https://github.com/rwjblue/ember-cli-ic-ajax/blob/master/index.js#L6-L8

is not resilient to dedupe induced topological shifts in the node_modules directory.

It is possible for NPM, to dedupe `ic-ajax` up the tree. NPM 3.0 does/will do this by default.

A not terrible nice, but easily improved solution example would be https://github.com/stefanpenner/ember-fetch/blob/master/index.js#L25-L30

**Comment by rwjblue** [comment]
Yes, we just need to use require.resolve. I was ignorant to many of the nesting related things in NPM when I originally wrote this code.

**Comment by stefanpenner** [comment]
Ya. Should be a quick fix sorta.

This was the last part of the default ember-cli setup that breaks in npm v3. So we will likely give people a nice tool/api for this.

**Comment by sir-dunxalot** [comment]
I think I'm getting stung by this. Running version 0.1.1, I got the following error when running `ember s`:

```
ENOENT: no such file or directory, scandir 'node_modules/ember-cli-ic-ajax/node_modules/'
```

I upgraded to the latest version of this addon (0.2.1) and `ember s` then caused the following error:

```
Path or pattern "vendor/ic-ajax/dist/named-amd/main.js" did not match any files
```

NPM: 3.3.3
Ember CLI: 0.2.7
Node: 4.0.0

**Comment by rwjblue** [comment]
Short answer: you can't use NPM 3 with ember-cli@0.2.7 (https://github.com/ember-cli/ember-cli/pull/4349 is required).

**Comment by erichonkanen** [comment]
hello, I am getting this on heroku using [ember-cli-heroku-buildpack](https://github.com/tonycoco/heroku-buildpack-ember-cli/)

`no such file or directory, scandir 'node_modules/ember-cli-ic-ajax/node_modules/'`

**Comment by chadian** [comment]
I also ran into a similar issue, I think this issue is similar to other `Error: ENOENT: no such file or directory` errors if you are running a npm version that's too new, as @rwjblue mentioned.

Double-check your npm version by:
`npm -v`

If you're on NPM 3, you can change your npm version globally by:
`npm install -g npm@2.14.3`

Now `npm -v` reports:
`2.14.3`

Removing the `node_modules` folder, then running `npm install`, then `ember s`, and everything is running smoothly.

**Comment by erichonkanen** [comment]
@chadian thanks! I was able to get past it with the answer in here https://github.com/tonycoco/heroku-buildpack-ember-cli/issues/97

**Comment by johnnyshields** [comment]
Guys, please get your shit together and fix this issue. It's been open for 6 months, this repo is a whopping 12-lines of code and thanks to your negligence here we spent hours debugging why our CI wouldn't deploy (we upgraded Node awhile back) and we missed a release deadline. We ultimately solved this by excising `ember-cli-ic-ajax` from our app entirely.

**Comment by johnnyshields** [comment]
Pardon my venting, but there's really no excuse for this sort of thing.

**Comment by rwjblue** [comment]
You are correct, there is no excuse for your verbal abuse.

**Comment by stefanpenner** [comment]
@johnnyshields please respect our community guidelines http://emberjs.com/guidelines/, just so you are aware your above tone is not something I would consider empathetic of respectful.

This issue appears to have been left open by mistake, and was addressed by https://github.com/rwjblue/ember-cli-ic-ajax/commit/c19b739e02ea147a60b2ee9d277e8a4b017aa775. 

As with all software, it is possible something regressed, if that is the case please provide a new issue with detailed and reproducible steps.

**Comment by johnnyshields** [comment]
@stefanpenner @rwjblue I apologize for my tone and I'm glad to see the issue is resolved.

These Ember CLI "glue" addons are quite brittle by nature. Would like to see first-class support for Browserify which would in many cases eliminate the need for glue libs.

**Comment by stefanpenner** [comment]
Our work on https://github.com/ember-cli/ember-cli/issues/4211 brings both browserify (for legacy) and esnext resolving as part of the default stack. We reprioritized to fix build performance, and once we land 2.x (which includes moving all primary dependencies away from bower and to NPM) we will resume effort on the linker/packager work, as the big blocking task for it was actually moving all primary dependencies from bower to npm.

**Comment by johnnyshields** [comment]
@stefanpenner that's wonderful to hear--build speed is certainly improving recently and really looking forward to npm-based package mgmt.

**Comment by stefanpenner** [comment]
@johnnyshields my above reference to the CoC isn't meant as a public shaming. Rather it is to convey that we respond well to constructive criticism/feedback/dialogue, but do not tolerate excessive negativity or otherwise hostile and derogatory remarks.

Unfortunately, your initial comments tend to take the later form, which makes these interactions quite taxing to say the least. I am unclear why that is, but please rest assured in this community we take polite, helpful concerns or criticisms very seriously. If anything, those interactions are both pleasant for all parties and much more motivating.

**Comment by johnnyshields** [comment]
@stefanpenner firstly I agree my tone was a bit overboard. That being said, here's a thread you and I had on Feb 8 about glue addons and my suggestion to give Browserify first-class adoption:
https://github.com/ember-cli/ember-cli/issues/2997#issuecomment-73373353

So you can understand why a bit I'm ripping my hair out 10 months later when Browserify was given a backseat to Ember Addons at the time, and we now have exactly the sorts of issues that always happen with glue libs in any framework.

**Comment by iorrah** [comment]
@chadian thanks, it worked for me.
